src/Grpc.NetCore.HttpClient/GrpcClientFactory.cs

```diff
@@ -0,0 +1,72 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Linq.Expressions;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using Grpc.Core;
+
+namespace Grpc.NetCore.HttpClient
+{
+    public static class GrpcClientFactory
+    {
+        public static TClient Create<TClient>(string baseAddress, X509Certificate certificate = null) where TClient : ClientBase<TClient>
+        {
+            // Needs to be set before creating the HttpClientHandler
+            AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2Support", true);
```

Comment:
Will we always need this? Will it have an affect on other HttpClient instances in the application?

Reply:
Yes, these are global. I imagine you won't always need it.

Reply:
This will be needed until the BCL turn this on by default. I don't think that's happening until at least the trailer support is available. cc @shirhatti.